mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/mapping/impl/PropertyIdentifierMapping.java

```diff
@@ -46,8 +50,7 @@ else if ( property != null ) {
 			return (I) caster.cast( id );
 		}
 		else {
-			throw new SearchException( "No identifier was provided, and this mapping does not define"
```

Comment:
Actually this code looks useless: `property` is always non-null. Could you remove this `else` branch and replace the `else if` above with an `else`? Then you can remove the `cannotExtractIdentifier`  method from the `Log` interface.

Reply:
May I remove the else branch at all here?